hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/mongodb/FieldResolver.java

```diff
@@ -160,16 +159,23 @@ private void validateType(MappingField field, DocumentField documentField) {
         }
     }
 
+    @Value.Immutable
+    interface ClientWithDataConnection {
+        MongoClient client();
+        @Nullable MongoDataConnection dataConnection();
+    }
+
     Map<String, DocumentField> readFields(String[] externalNames,
                                           String dataConnectionName,
                                           Map<String, String> options,
                                           boolean stream) {
-        String collectionName = externalNames[0]; // TODO HZ-2260
+        String collectionName = externalNames.length == 2 ? externalNames[1] : externalNames[0];
+        String databaseName = Options.getDatabaseName(nodeEngine, externalNames, dataConnectionName);
+        ClientWithDataConnection connect = connect(dataConnectionName, options);
```

Comment:
You actually don't need both client and dataConnection. It's enough in `connect` to do the following:

- get dataConnection 
- get client
- release the dataConnection
- close the client

You don't need to keep the dataConnection until you close the client.